### ISSUE TITLE
[Bug] AWQ量化InternVL2 40B输出无意义的结果

### ISSUE THREAD

**Issue by lijing1996** [InternLM/lmdeploy#2017]
### Checklist

- [X] 1. I have searched related issues but cannot get the expected help.
- [X] 2. The bug has not been fixed in the latest version.

### Describe the bug

AWQ量化InternVL2 40B输出无意义的结果。我自己量化了8B、26B结果都是正常的。40B的结果输出就是乱码。

输出如下
thanks Forbes Forbes Forbes Forbes Boulder Boulder Boulder slideshow slideshowTimestamp vast vast vast automation automation automation automati
on automation automation automation testimony testimony testimony parameter Forbes Forbes slideshow Forbes documentation documentation刘邦 Mg arriving Analysis Analysis slideshow slideshow slideshow assert assert assert assert transc
ripts documentation documentation slideshow slideshow handheld slideshow slideshow slideshow documentation documentation automation automation automation automation automation automation automation automation automation automation au
tomation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation auto
mation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automa
tion automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automati
on automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation
 automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation a
utomation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation aut
omation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation autom
ation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automat
ion automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automatio
n automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation 
automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation au
tomation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation auto
mation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation automation 

### Reproduction

lmdeploy lite auto_awq  InternVL2-40B --work-dir InternVL2-40B-4bit

### Environment

```Shell
sys.platform: linux
Python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
CUDA available: True
MUSA available: False
numpy_random_seed: 2147483648
GPU 0: NVIDIA H800
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 12.3, V12.3.107
GCC: x86_64-linux-gnu-gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
PyTorch: 2.2.0a0+81ea7a4
PyTorch compiling details: PyTorch built with:
  - GCC 11.2
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2021.1-Product Build 20201104 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v3.1.1 (Git Hash N/A)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 12.3
  - NVCC architecture flags: -gencode;arch=compute_52,code=sm_52;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_72,code=sm_72;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_87,code=sm_87;-gencode;arch=compute_90,code=sm_90;-gencode;arch=compute_90,code=compute_90
  - CuDNN 8.9.7  (built against CUDA 12.2)
  - Magma 2.6.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=12.3, CUDNN_VERSION=8.9.7, CXX_COMPILER=/opt/rh/gcc-toolset-11/root/usr/bin/c++, CXX_FLAGS=-fno-gnu-unique -D_GLIBCXX_USE_CXX11_ABI=1 -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=range-loop-construct -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-stringop-overflow -Wsuggest-override -Wno-psabi -Wno-error=pedantic -Wno-error=old-style-cast -Wno-missing-braces -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.2.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=ON, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

TorchVision: 0.17.0a0
LMDeploy: 0.5.0+4c8769e
transformers: 4.37.2
gradio: 3.50.2
fastapi: 0.110.3
pydantic: 2.5.3
triton: 2.1.0
NVIDIA Topology: 
        GPU0    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      SYS     SYS     PIX     SYS     NODE    NODE    SYS     NODE    0-85    0               N/A
NIC0    SYS      X      NODE    SYS     NODE    SYS     SYS     NODE    SYS
NIC1    SYS     NODE     X      SYS     NODE    SYS     SYS     NODE    SYS
NIC2    PIX     SYS     SYS      X      SYS     NODE    NODE    SYS     NODE
NIC3    SYS     NODE    NODE    SYS      X      SYS     SYS     NODE    SYS
NIC4    NODE    SYS     SYS     NODE    SYS      X      NODE    SYS     NODE
NIC5    NODE    SYS     SYS     NODE    SYS     NODE     X      SYS     NODE
NIC6    SYS     NODE    NODE    SYS     NODE    SYS     SYS      X      SYS
NIC7    NODE    SYS     SYS     NODE    SYS     NODE    NODE    SYS      X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: 
  NIC1: 
  NIC2: 
  NIC3: 
  NIC4: 
  NIC5: 
  NIC6: 
  NIC7:
```

**Comment by zhyncs** [comment]
ref https://github.com/InternLM/lmdeploy/issues/2009#issuecomment-2225099462 cc @AllentDan

**Comment by AllentDan** [comment]
Fixed in #2023
